### PR TITLE
Start to document Storyscript's type system

### PR DESCRIPTION
**Pull request by wilzbach**
This is a first start at documenting what's allowed. Any suggestions are welcome!

The table doesn't look super-nice yet:

image

CC @judepereira

### CODE REVIEW

**Comment by JeanBarriere** [comment]
@wilzbach I forgot the style for the headers when I refactored the style, just pushed a fix. You can rebase from master so you can see it perfectly in the netlify preview :)

**Comment by wilzbach** [comment]
Thanks! Looks a lot better :+1: 


image

**Comment by judepereira** [comment]
@wilzbach I like it. Let's add a note on automatic type casting? Such as int + float will return a float (cast to upper type).

File: storyscript/README.md [hunk 1]
```diff
@@ -738,20 +738,95 @@ Storyscript allows a few implicit type conversions:
...
+
+Type       | Operations | Remarks
+-----------|------------|------
+`boolean`  | all        | Operations between two `boolean`s are implicitly converted to `int`
+`int`      | all        |
+`float`    | all        |
+`regexp`   | (none)     |
+`time`     | `+`, `-`   |
+`string`   | `+`        | Addition with any other type is possible, returns a `string`
+`list`     | `+`        |
```
Comment (judepereira) [inline storyscript/README.md]:
```suggestion
`List`     | `+`        |
```

File: storyscript/README.md [hunk 2]
```diff
@@ -738,20 +738,95 @@ Storyscript allows a few implicit type conversions:
...
+`map`      | (none)     |
```
Comment (judepereira) [inline storyscript/README.md]:
```suggestion
`Map`      | (none)     |
```

**Comment by wilzbach** [comment]
> I like it. Let's add a note on automatic type casting? Such as int + float will return a float (cast to upper type).

Already there: https://docs.asyncy.com/storyscript/#type-checking

**Pull request merged**